arangod/Aql/IResearchViewExecutor.tpp

```diff
@@ -824,6 +721,55 @@ IResearchViewExecutorBase<Impl, ExecutionTraits>::skipRowsRange(
   return {reportedState, stats, call.getSkipCount(), upstreamCall};
 }
 
+template<typename Impl, typename ExecutionTraits>
+void IResearchViewExecutorBase<Impl, ExecutionTraits>::materialize() {
+  // TODO(MBkkt) unify it more with MaterializeExecutor
+  auto pkReadingOrder = _indexReadBuffer.getMaterializeRange(0);
```

Comment:
Could we avoid using this method here? And directly only do `iota` for merge executors? Custom materialize range makes sense only for heap sort. So hardcoded call with 0  skip is really overkill. For simple executor that is also an overkill as entire buffer could be processed as is and `it` could be just a counter.

Reply:
We can avoid this allocation only for simple view executor (not heap or merge).
(Because only this executor have order of produced documents that corresponding to snapshot)

But it needs to make quite a lot constexpr code.
So I think about make it (todo comment in getMaterializeRange) after fix possible ping pong between rocksdb and search.
And stabilize code more

Reply:
BTW For merge sort we can keep this vector as a member to avoid reallocations multiple times. As most likely all batches except last one would have same size

Reply:
For heap we also can, just needs to create another buffer.

I think it should be in the MultiGet context:
Something like this:
```
union InputOutput {
  void* input;         // pointer to the localdocid + segment
  uint8_t* output;  // allocated output data, could be null
};

std::vector<InputOutput> _values;

~MultiGetContext() {
  for (size_t i = 0; i != _global; ++i) delete[] _values[i].output;
}

std::unique_ptr<uint8_t[]> extract(size_t i) {
  return {std::exchange(_values[i], nullptr)};
}
```

Reply:
In such case we never need additional allocation

Reply:
As I mentioned I'm aware about it but want to do it in the next step of materialize improvements

Reply:
Ok. Let's merge this PR then in this state. And make more performance improvements in the next steps.